src/Nethermind/Nethermind.Core/SlotTime.cs

```diff
@@ -2,13 +2,12 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
-using Nethermind.Core;
 
-namespace Nethermind.Shutter;
+namespace Nethermind.Core;
 
-public class ShutterTime(ulong genesisTimestampMs, ITimestamper timestamper, TimeSpan slotLength, TimeSpan blockUpToDateCutoff)
+public class SlotTime(ulong genesisTimestampMs, ITimestamper timestamper, TimeSpan slotLength, TimeSpan blockUpToDateCutoff)
```

Comment:
```suggestion
public readonly struct SlotTime(ulong genesisTimestampMs, ITimestamper timestamper, TimeSpan slotLength, TimeSpan blockUpToDateCutoff)
```

or at least `readonly class`

Reply:
If we go with readonly struct maybe pass it with `in` parameter where possible?

Reply:
If we did this we may as well make a class right? Also since it's just one `ulong` I think it makes sense to pass by value, the overhead could be greater from indirection when passing by reference

Reply:
well it is also ITimestamper reference  and 2 timespans

Reply:
Ah right I can make a new PR. Is there any reason not to make a class?

Reply:
Ah right I can make a new PR. Is there any reason not to make a class?